src/util.js

```diff
@@ -84,63 +84,81 @@ function getLast(arr) {
   return null;
 }
 
-function skipNewLineForward(text, index) {
-  // What the "end" location points to is not consistent in parsers.
-  // For some statement/expressions, it's the character immediately
-  // afterward. For others, it's the last character in it. We need to
-  // scan until we hit a newline in order to skip it.
-  while (index < text.length) {
-    if (text.charAt(index) === "\n") {
-      return index + 1;
+function skip(chars) {
+  return (text, index, backwards) => {
+    // Allow `skip` functions to be threaded together without having
+    // to check for failures (did someone say monads?).
+    if(index === false) {
+      return false;
     }
-    if (text.charAt(index) === "\r" && text.charAt(index + 1) === "\n") {
-      return index + 2;
+
+    const length = text.length;
+    let cursor = index;
+    while (cursor >= 0 && cursor < length) {
+      const c = text.charAt(cursor);
+      if(chars.test) {
```

Comment:
`chars instanceof RegExp` would be clearer imo

![image](https://cloud.githubusercontent.com/assets/197597/22344985/2f5d968a-e3b3-11e6-86a3-b0e6ce560bd4.png)